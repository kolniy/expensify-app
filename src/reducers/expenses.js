// Expense Reducer
const expenseReducerDefault = []
export default (state = expenseReducerDefault, action ) => {
    switch(action.type){
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ];
        case 'REMOVE_EXPENSE':
                return state.filter(({ id }) => id !== action.id )
        case 'EDIT_EXPENSE':
            // return state.map((expense) => {
                // if(expense.id === action.id){
                //     return {
                //         ...expense,
                //         ...action.updates
                //     }
                // } else {
                //     expense
                // }              
            // })
            var arrayWithEditedItem = []
            for (var i = 0; i < state.length; i++){
                if(state[i].id === action.id){
                    arrayWithEditedItem.push({
                        ...state[i],
                        ...action.updates
                    })
                } else {
                    arrayWithEditedItem.push(state[i])
                }
            }
            return arrayWithEditedItem
        default:
            return state    
    }
}