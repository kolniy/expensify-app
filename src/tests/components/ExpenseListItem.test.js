import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'

test('Render expense list with fixtures data', () => {
    const wrapper = shallow(<ExpenseListItem key={expenses[1].id} {...expenses[1]}/>)
    expect(wrapper).toMatchSnapshot()
})