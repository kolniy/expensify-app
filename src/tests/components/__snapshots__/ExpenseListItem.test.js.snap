// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render expense list with fixtures data 1`] = `
<Link
  className="list-item"
  to="/edit/2"
>
  <div>
    <h3
      className="list-item__title"
    >
      Rent
    </h3>
    <span
      className="list-item__sub-title"
    >
      December 28th, 1969
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $1,095.00
  </h3>
</Link>
`;
